src/main/java/wolox/training/models/User.java

```diff
@@ -0,0 +1,132 @@
+package wolox.training.models;
+
+import static wolox.training.utils.ErrorConstants.BOOK_ALREADY_OWNED;
+
+import com.fasterxml.jackson.annotation.JsonIdentityInfo;
+import com.fasterxml.jackson.annotation.ObjectIdGenerators;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.ManyToMany;
+import javax.persistence.SequenceGenerator;
+import javax.persistence.Table;
+import javax.validation.constraints.NotNull;
+import wolox.training.exceptions.BookAlreadyOwnedException;
+
+@Entity
+@Table(name = "t_user")
+@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
+public class User {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "USER_SEQ")
+    @SequenceGenerator(name = "USER_SEQ", sequenceName = "USER_SEQ")
+    private Long id;
+
+    @NotNull
+    @Column(nullable = false)
+    private String username;
+
+    @NotNull
+    @Column(nullable = false)
+    private String name;
+
+    @NotNull
+    @Column(nullable = false)
+    private LocalDate birthdate;
+
+    @NotNull
+    @Column(nullable = false)
+    @ManyToMany(cascade = {CascadeType.PERSIST, CascadeType.REFRESH})
+    private List<Book> books = new ArrayList<>();
+
+    public User() {}
+
+    public Long getId() {
+        return id;
+    }
+
+    public String getUsername() {
+        return username;
+    }
+
+    public void setUsername(String username) {
+        this.username = username;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public LocalDate getBirthdate() {
+        return birthdate;
+    }
+
+    public void setBirthdate(LocalDate birthdate) {
+        this.birthdate = birthdate;
+    }
+
+    public List<Book> getBooks() {
+        return Collections.unmodifiableList(books);
+    }
+
+    public void setBooks(List<Book> books) {
+        this.books = books;
+    }
+
+    public void addBook(Book book) {
+        if (books.contains(book)) {
+            throw new BookAlreadyOwnedException(BOOK_ALREADY_OWNED);
+        } else {
+            this.books.add(book);
+        }
+    }
+
+    public void removeBook(Book book) {
+        if (books.contains(book)) this.books.remove(book);
```

Comment:
We could add an exception like you did in addBook, to give it a behavior in case the book doesn't exist. What do you say?

Reply:
Sound good. I'll add it